src/istgt_lu_disk.c

```diff
@@ -5668,17 +5658,13 @@ check_for_command_completion(spec_t *spec, rcommon_cmd_t *rcomm_cmd, ISTGT_LU_CM
 		} else {
 			data = handle_read_consistency(rcomm_cmd);
 			cmd->data = data;
-			cmd->data_len = rcomm_cmd->data_len;
 			rc = 1;
 		}
 	} else if (rcomm_cmd->opcode == ZVOL_OPCODE_WRITE) {
 		if (success >= spec->consistency_factor) {
```

Comment:
let us store consistency_factor also in rcomm_cmd, so that, any change in it won't be the problem.

Reply:
consistency_factor is protected through spec level mutex. If we store consistency_factor in rcomm_cmd, we won't be able to reflect the change in consistency factor to old queued command.